Firestore/Source/Local/FSTLocalStore.mm

```diff
@@ -280,6 +290,7 @@ - (MaybeDocumentMap)applyRemoteEvent:(FSTRemoteEvent *)remoteEvent {
           (authoritativeUpdates.contains(doc.key) && !existingDoc.hasPendingWrites) ||
           doc.version >= existingDoc.version) {
         [self.remoteDocumentCache addEntry:doc];
```

Comment:
Profiler shows that currently ~30% of time is spent here, or more specifically, [here](https://github.com/firebase/firebase-ios-sdk/blob/master/Firestore/core/src/firebase/firestore/local/leveldb_transaction.h#L165):
```cpp
void Put(absl::string_view key, GPBMessage* message) {
    NSData* data = [message data]; // THIS LINE; apparently, `data` is pretty time-consuming
    std::string key_string(key);
    mutations_[key_string] = std::string((const char*)data.bytes, data.length);
    version_++;
  }
```
I can't see a way to improve this immediately, but I won't be surprised if the nanopb-based serializer will bring a further performance improvement.